lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1337,6 +1341,23 @@ class HSADevice final : public KalmarDevice
         }
 
         return checksum.str();
+#else
+        // FNV-1a hashing, 64-bit version
+        const uint64_t FNV_prime = 0x100000001b3;
+        const uint64_t FNV_basis = 0xcbf29ce484222325;
+        uint64_t hash = FNV_basis;
+
+        const char *str = static_cast<const char *>(source);
+
+        // 104 is the proper size from Jack's research
```

Comment:
So the gpu_deconvolution work is actually using LC backend (due to hcFFT). If I use the original size in FNV-loop, the execution time is about 2m37s.

Here's the comparison of each condition:

- original: 3m10s
- FNV hash: 2m37s
- FNV hash + 104 cutoff: 1m59s

Personally I would prefer having FNV hash with some fixed size cutoff for the best performance.